### PR TITLE
CLN: Period tests

### PR DESCRIPTION
**Pull request by jbrockmendel**
Pretty much re-wrote the comparison tests to match the patterns we use elsewhere, avoided bare pytest.raises.

A bunch of tests use `Period("NaT", freq)` which ends up being very duplicative.  Cut that down a bit and marked a couple tests that belong elsewhere, will move in a separate pass.

### CODE REVIEW

File: pandas/tests/scalar/period/test_period.py [hunk 1]
```diff
@@ -1281,90 +1278,39 @@ def test_add_offset(self):
...
-            for o in [
-                offsets.YearBegin(2),
-                offsets.MonthBegin(1),
-                offsets.Minute(),
-                np.timedelta64(3200, "s"),
-                timedelta(hours=23, minutes=30),
-            ]:
-                assert p + o is NaT
-                assert o + p is NaT
+        for o in [
```
Comment (jreback) [inline pandas/tests/scalar/period/test_period.py]: can you parameterize while you are at it

**Review by jreback — CHANGES_REQUESTED** [review]
small comment, worth doing in this PR.

File: pandas/tests/scalar/period/test_period.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/scalar/period/test_period.py]: parametrized+green

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks

**Pull request merged**